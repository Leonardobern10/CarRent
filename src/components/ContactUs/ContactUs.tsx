import type { ReactElement } from "react";
import TitleSection from "../shared/TitleSection";
import ButtonDefault from "../shared/ButtonDefault";
import keyImage from "../../assets/images/key_image.png";
import { motion } from "motion/react";

export default function ContactUs(props: { title: string }): ReactElement {
  return (
    <section className="row-span-3 lg:row-span-5 col-span-7 lg:bg-linear-to-t lg:from-60% lg:from-black-medium lg:to-60% lg:to-white-background flex flex-col gap-y-4 lg:gap-y-20 justify-evenly items-center font-poppins">
      <TitleSection text={props.title} />
      <motion.div
        initial={{ opacity: 0 }}
        whileInView={{ opacity: 1 }}
        transition={{ ease: "backOut", duration: 1.3 }}
        className="flex flex-row h-fit lg:h-5/7 w-full lg:w-5/6 bg-white-background max-lg:pb-12"
      >
        <form
          data-testid="contact-form"
          action=""
          className="w-full h-fit lg:h-5/6 lg:w-1/2 flex flex-col justify-center items-center gap-y-10"
        >
          <input
            type="text"
            placeholder="Seu nome*"
            className="input"
            required
          />
          <input
            type="text"
            placeholder="Seu e-mail*"
            className="input"
            required
          />
          <textarea
            name=""
            id=""
            placeholder="Sua mensagem"
            className="input"
          />
          <ButtonDefault
            buttonName="Enviar mensagem"
            width="1/2"
            height="10"
            weight="normal"
            fontSize="sm"
          />
        </form>
        <div className="max-lg:hidden w-1/2 lg:h-5/6">
          <img
            src={keyImage}
            className="w-full h-full"
            alt="Recebendo a chave do carro"
          />
        </div>
      </motion.div>
    </section>
  );
}
