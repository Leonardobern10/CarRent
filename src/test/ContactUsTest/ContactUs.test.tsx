import ContactUs from "../../components/ContactUs/ContactUs";
import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

const titleTest = "Titulo de teste.";
const buttonName = "Enviar mensagem";

describe("ContactUs", () => {
  it("renderiza corretamente o componente <TitleSection />", () => {
    render(<ContactUs title={titleTest} />);
    expect(screen.getByText(titleTest)).toBeInTheDocument();
  });

  it("renderiza o formulário com as classes esperadas", () => {
    render(<ContactUs title={titleTest} />);
    const form = screen.getByTestId("contact-form");

    const expectedClasses = [
      "w-full",
      "h-fit",
      "lg:h-5/6",
      "lg:w-1/2",
      "flex",
      "flex-col",
      "justify-center",
      "items-center",
      "gap-y-10",
    ];

    expectedClasses.forEach((cls) => {
      expect(form.className).toContain(cls);
    });
  });

  it("renderiza os campos do formulário corretamente", () => {
    render(<ContactUs title={titleTest} />);
    expect(screen.getByPlaceholderText("Seu nome*")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Seu e-mail*")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Sua mensagem")).toBeInTheDocument();
  });

  it("renderiza o botão de envio", () => {
    render(<ContactUs title={titleTest} />);
    expect(
      screen.getByRole("button", { name: buttonName })
    ).toBeInTheDocument();
  });

  it("renderiza a imagem da seção", () => {
    render(<ContactUs title={titleTest} />);
    const img = screen.getByRole("img");
    expect(img).toBeInTheDocument();
  });
});
