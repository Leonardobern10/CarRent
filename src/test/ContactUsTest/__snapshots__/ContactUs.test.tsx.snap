// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactUs > faz snapshot do componente 1`] = `
<DocumentFragment>
  <section
    class="row-span-3 lg:row-span-5 col-span-7 lg:bg-linear-to-t lg:from-60% lg:from-black-medium lg:to-60% lg:to-white-background flex flex-col gap-y-4 lg:gap-y-20 justify-evenly items-center font-poppins"
  >
    <div
      id="title-section"
    >
      <h2
        class="text-xl md:text-2xl lg:text-4xl font-montserrat pb-3 font-medium text-black-medium  decoration-orange-primary"
      >
        Titulo de teste.
      </h2>
      <div
        class="w-full h-[3px] bg-orange-primary"
      />
    </div>
    <div
      class="flex flex-row h-fit lg:h-5/7 w-full lg:w-5/6 bg-white-background max-lg:pb-12"
      style="opacity: 0;"
    >
      <form
        action=""
        class="w-full h-fit lg:h-5/6 lg:w-1/2 flex flex-col justify-center items-center gap-y-10"
        data-testid="contact-form"
      >
        <input
          class="input"
          placeholder="Seu nome*"
          required=""
          type="text"
        />
        <input
          class="input"
          placeholder="Seu e-mail*"
          required=""
          type="text"
        />
        <textarea
          class="input"
          id=""
          name=""
          placeholder="Sua mensagem"
        />
        <button
          class="bg-orange-primary rounded-sm p-2 text-sm font-normal text-black-medium font-poppins font-bold text-black-medium min-w-30 w-1/2 h-10 hover:cursor-pointer hover:bg-black-medium hover:text-bege transition-colors"
        >
          Enviar mensagem
        </button>
      </form>
      <div
        class="max-lg:hidden w-1/2 lg:h-5/6"
      >
        <img
          alt="Recebendo a chave do carro"
          class="w-full h-full"
          src="/src/assets/images/key_image.png"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
